docs/rules/custom-error.md

```diff
@@ -0,0 +1,96 @@
+# Enforce correct Error subclassing
+
+Enforces the only valid way of `Error` subclassing. It works with any super class that ends in `Error`.
+
+
+## Fail
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+		this.message = message;
+		this.name = 'CustomError';
+	}
+}
+```
+
+The `this.message` assignment is useless as it's already set via the `super()` call.
+
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super();
+		this.message = message;
+		this.name = 'CustomError';
+	}
+}
+```
+
+Pass the error message to `super()` instead of setting `this.message`.
+
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+	}
+}
+```
+
+No `name` property set. The name property is needed so the error shows up as `[CustomError: foo]` and not `[Error: foo]`.
+
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+		this.name = this.constructor.name;
+	}
+}
+```
+
+Use a string literal to set the `name` property as it will not change after minifying.
+
+
+```js
+class CustomError extends Error {
+	constructor(message) {
+		super(message);
+		this.name = 'MyError';
+	}
+}
+```
+
+The `name` property should be set to the class name.
+
+
```

Comment:
Can you add an example of invalid class name?

Per the `Invalid class name, use ${className}` violation.